incubator/virtualcluster/pkg/syncer/resources/pod/controller.go

```diff
@@ -143,6 +164,72 @@ func (c *controller) enqueuePod(obj interface{}) {
 	c.queue.Add(key)
 }
 
+// c.Mutex needs to be Locked before calling addToClusterVNodeGCMap
+func (c *controller) addToClusterVNodeGCMap(cluster string, nodeName string) {
+	if _, exist := c.clusterVNodeGCMap[cluster]; !exist {
+		c.clusterVNodeGCMap[cluster] = make(map[string]VNodeGCStatus)
+	}
+	c.clusterVNodeGCMap[cluster][nodeName] = VNodeGCStatus{
+		QuiesceStartTime: metav1.Now(),
+		Phase:            VNodeQuiescing,
+	}
+}
+
+// c.Mutex needs to be Locked before calling removeQuiescingNodeFromClusterVNodeGCMap
+func (c *controller) removeQuiescingNodeFromClusterVNodeGCMap(cluster string, nodeName string) bool {
+	if _, exist := c.clusterVNodeGCMap[cluster]; exist {
+		if _, exist := c.clusterVNodeGCMap[cluster][nodeName]; exist {
+			if c.clusterVNodeGCMap[cluster][nodeName].Phase == VNodeQuiescing {
+				delete(c.clusterVNodeGCMap[cluster], nodeName)
+				return true
+			} else {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func (c *controller) updateClusterVNodePodMap(cluster string, vPod *v1.Pod, event reconciler.EventType) {
+	nodeName := vPod.Spec.NodeName
+	if nodeName == "" || !isPodScheduled(vPod) {
+		return
+	}
+	func() {
+		c.Lock()
+		defer c.Unlock()
+		if event == reconciler.AddEvent || event == reconciler.UpdateEvent {
+			if _, exist := c.clusterVNodePodMap[cluster]; !exist {
+				c.clusterVNodePodMap[cluster] = make(map[string]map[string]struct{})
+			}
+			if _, exist := c.clusterVNodePodMap[cluster][nodeName]; !exist {
+				c.clusterVNodePodMap[cluster][nodeName] = make(map[string]struct{})
+			}
+			c.clusterVNodePodMap[cluster][nodeName][vPod.Name] = struct{}{}
```

Comment:
pod name is unique among each namespace?

Reply:
Nice catch. Fixed.